### PR TITLE
Expose the platform directly for apple products

### LINKED ISSUE #53579: MonoVM: Missing runtime workloads for Mac Catalyst and macOS
### Description

Currently there are runtime workloads for iOS and tvOS:

https://github.com/dotnet/runtime/blob/dbb05eba3b229df3d4c15ce08de9f84c7febb7b3/src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.json.in#L33-L63

but we're missing runtime workloads for Mac Catalyst and macOS.

While not strictly necessary, this would make it easier for us to acquire the AOT compilers for Mac Catalyst and macOS.

CC @steveisok

### LINKED ISSUE #54996: Microsoft.NET.Workload.Mono.Toolchain does not get me the arm64 AOT compiler for Mac Catalyst
### Description

If I try to use the AOT compiler for Mac Catalyst/ARM64 (RID: `maccatalyst-arm64`), it doesn't work, because the only aot compiler in the MonoAotCrossCompiler item group is the one for `maccatalyst-x64`:

Screenshot 2021-07-01 at 09 24 01

This is when executing on an M1 machine (but with the x64 version of .NET)

Binlog: [msbuild.binlog.zip](https://github.com/dotnet/runtime/files/6746193/msbuild.binlog.zip)

CC @steveisok @akoeplinger

### PR DESCRIPTION
**Pull request by lewing**
Fixes https://github.com/dotnet/runtime/issues/54996

Fixes https://github.com/dotnet/runtime/issues/53579

### CODE REVIEW

File: src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets [hunk 1]
```diff
@@ -25,25 +25,30 @@
...
     </ImportGroup>
 
     <ImportGroup Condition="'$(TargetPlatformIdentifier)' == 'tvos' and $([MSBuild]::IsOSPlatform('osx'))">
         <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
         <Import Project="Sdk.props" Sdk="Microsoft.NETCore.App.Runtime.AOT.Cross.tvos-arm64" />
+        <Import Project="Sdk.props" Sdk="Microsoft.NETCore.App.Runtime.AOT.Cross.tvos-x64" />
         <Import Project="Sdk.targets" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
     </ImportGroup>
 
     <ImportGroup Condition="'$(TargetPlatformIdentifier)' == 'tvossimulator'">
```
Comment (steveisok) [inline src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets:48]: I'll tweak it - `tvossimulator` entries need to go up under `tvos`

Comment (lewing) [reply src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets]: fixed

**Comment by lewing** [comment]
all infrastructure failures

**Comment by steveisok** [comment]
@rolfbjarne With this change, you'll need to remove the RuntimeConfig imports that are in your `WorkloadManifest.targets` files.  Also, what runtime are we defaulting Xam.Mac to?  Same question I asked in https://github.com/dotnet/runtime/issues/53579#issuecomment-877914269

**Review by rolfbjarne — APPROVED** [review]

**Review by jonathanpeppers — APPROVED** [review]

File: src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in [hunk 2]
```diff
@@ -54,4 +73,43 @@
         <Import Project="Sdk.targets" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
     </ImportGroup>
 
+    <PropertyGroup Condition="'$(TargetPlatformIdentifier)' == 'android' or '$(TargetPlatformIdentifier)' == 'macOS' or '$(TargetPlatformIdentifier)' == 'ios' or '$(TargetPlatformIdentifier)' == 'maccatalyst' or '$(TargetPlatformIdentifier)' == 'tvos' or ('$(RuntimeIdentifier)' == 'browser-wasm' and '$(UsingBrowserRuntimeWorkload)' == 'true')">
```
Comment (lewing) [inline src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in:76]: Is this a strict enough check on macOS?

Comment (steveisok) [reply src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in:76]: I thought it wasn't at first, but I believe macOS is the TFM for xam.mac.  So, we wouldn't be interfering w/ anything else.

**Comment by steveisok** [comment]
Failure is helix instance not having openssl installed and is unrelated.

File: src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in [hunk 3]
```diff
@@ -54,4 +73,43 @@
...
+        tvossimulator-x64;
+        android-arm64;
+        android-arm;
+        android-x64;
+        android-x86;
+        " />
+      
+      <KnownRuntimePack Remove="Microsoft.NETCore.App" />
+      <KnownRuntimePack Include="Microsoft.NETCore.App"
+                        TargetFramework="net6.0"
```
Comment (akoeplinger) [inline src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in:106]: @lewing @steveisok while this works for now, it'll break once we move to `net7.0`. Can you try whether doing this instead works:

```xml
<KnownRuntimePack Update="Microsoft.NETCore.App"
                  LatestRuntimeFrameworkVersion="$(_MonoWorkloadRuntimePackPackageVersion)"
                  Condition="'%(RuntimePackLabels)' == 'Mono' and '%(TargetFramework)' == 'net$(TargetFrameworkVersion)'"
```

Comment (steveisok) [reply src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in:106]: Sure thing

**Pull request merged**